sitemap_utils.py

```
@@ -0,0 +1,367 @@
+            if line.lower().startswith("sitemap:"):
+                sitemap_url = line[8:].strip()
+                sitemap_urls.append(sitemap_url)
+                
+        if sitemap_urls:
+            logger.info(f"Found {len(sitemap_urls)} sitemaps in robots.txt")
+        else:
+            logger.info("No sitemaps found in robots.txt")
+            
+        return sitemap_urls
+
+    def _parse_sitemap_xml(self, content: str) -> Tuple[List[SitemapURL], List[str]]:
```

Ruff (0.8.2)

`_parse_sitemap_xml` is too complex (11 > 10)

(C901)